doc-content/drools-docs/src/main/asciidoc/DecisionEngine/inference-and-truth-maintenance.adoc

```diff
@@ -0,0 +1,271 @@
+[id='inference-and-truth-maintenance_{context}']
+
+= Inference and truth maintenance in the {DECISION_ENGINE}
+
+The basic function of the {DECISION_ENGINE} is to match production facts and data to rules and determine whether and how rules are executed. To ensure that relevant data is applied to the appropriate rules, the {DECISION_ENGINE} makes inferences based on existing knowledge and performs the actions based on the inferred information.
+
+For example, the following DRL rule determines the age requirements for adults, such as in a bus pass policy:
+
+.Rule to define age requirement
+[source]
+----
+rule "Infer Adult"
+when
+  $p : Person(age >= 18)
+then
+  insert(new IsAdult($p))
+end
+----
+
+Based on this rule, the {DECISION_ENGINE} infers whether a person is an adult or a child and performs the specified action (the `then` consequence). Every person who is 18 years old or older has an instance of `IsAdult` inserted for them in the working memory. This inferred relation of age and bus pass can then be invoked in any rule, such as in the following rule segment:
+
+[source]
+----
+$p : Person()
+IsAdult(person == $p)
+----
+
+In many cases, new data in a rule system is the result of other rule executions, and this new data can affect the execution of other rules. If the {DECISION_ENGINE} asserts data as a result of executing a rule, the {DECISION_ENGINE} uses truth maintenance to justify the assertion and enforce truthfulness when applying inferred information to other rules. Truth maintenance also helps to identify inconsistencies and to handle contradictions. For example, if two rules are executed and result in a contradictory action, the {DECISION_ENGINE} chooses the action based on assumptions from previously calculated conclusions.
+
+The {DECISION_ENGINE} inserts facts using either stated or logical insertions:
+
+* *Stated insertions:* Defined with `insert()`. After stated insertions, facts are generally retracted explicitly.
+* *Logical insertions:* Defined with `insertLogical()`. After logical insertions, the facts that were inserted are automatically retracted when the conditions that inserted the facts are no longer true. The facts are retracted when no condition supports the logical insertion.
+
+The following flowcharts illustrate the life cycle of stated and logical insertions:
+
+.Stated insertion
+image::HybridReasoning/Stated_Assertion.png[]
+
+.Logical insertion
+image::HybridReasoning/Logical_Assertion.png[]
```

Comment:
the term JUSTIFIED should be at least defined somewhere. Otherwise both diagrams looks good, but I am still convinced they should be shown at the end.

Reply:
Done. Moved to the end, and added a paragraph on justification. See my last comment.